solution.cpp

```diff
@@ -1,134 +1,157 @@
-                  const string& infilename,
-                  const string& outfilename );
-
-/**
- * Read numbers from `infilename`, and if they are composite, output them to
- * `outfilename`, separated by newlines.
- *
- * Arguments:
- * - `size`: The size of (the number of elements in) the array.
- * - `primes`: An array of boolean values, with each element set to `true` if
- *   its index is prime, and `false` otherwise.
- * - `infilename`: A string representing the file path to read from.
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_composites( const int size,
-                      bool primes[],
-                      const string& infilename,
-                      const string& outfilename );
-
-// ----------------------------------------------------------------------------
-
-int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
-        return 1;  // error
-    }
-
-    bool primes[max+1];
-    sieve(max+1, primes);
-
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
-    }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
-    }
-
-    return 0;  // success
+//Find maximum unsigned integer within text file
+int find_max(const char* infilename);
+
+//Mark all prime indices in 'primes' as 'true'
+void sieve(bool primes[], const int max);
+
+//Read numbers from textfile, if they are prime, output them to a new text file 'primes'
+int write_primes(bool primes[], const int max,
+				 const char* infilename,
+				 const char* outfilename);
+
+//Read numbers from textfile, if they are composites, output them to a new text file 'composites'
+int write_composites(bool primes[], const int max,
+					 const char* infilename,
+					 const char* outfilename);
+
+int main ()
+{
+	
+	int max = find_max("input.txt"); //Calls find_max function
+
+	bool primes[max+1];
+	
+	sieve(primes,max+1); //Calls the sieve function
+
+	int ret; //Stores return values, checks for error code
+
+	ret =write_primes(primes,max+1, "input.txt","primes.txt");
+	if (ret == -1) 
+	{	cout << "ERROR in 'write_primes()" << endl;
+		return 1; //error
+	}
+
+	ret = write_composites(primes,max+1, "input.txt","composites.txt");
+	if (ret == -1) 
+	{	cout << "ERROR in 'write_primes()" << endl;
+		return 1; //error
+	}
+
+	return 0; //success
 }
 
-// ----------------------------------------------------------------------------
+int find_max(const char* infilename)
+{
+	fstream file;
+	int num[8];
+
+	file.open(infilename,ios::in);
+
+	for (int i=0;i<8;i++)
+	{
+		file >> num[i];
+	}
```

Comment:
you shouldn't need to store all the numbers at once -- it should be possible to process them as they're being read in